src/session.rs

```diff
@@ -80,12 +81,14 @@ impl Session {
         let mut url_parser = UrlParser::new();
         url_parser.scheme_type_mapper(scheme_type_mapper);
         let url = try!(url_parser.parse(url_string));
+        let tls = if url.scheme == "amqps" { true } else { false };
```

Comment:
I find it too confusing/implicit, but I can settle on
`let tls = { url.scheme == "amqps" };`